### ISSUE TITLE
Setup and evaluate MyTardis search

### ISSUE THREAD

**Issue by craigpatten** [modc08/modc08#6]
https://mytardis.readthedocs.org/en/latest/searchsetup.html

**Comment by millercathy** [comment]
Hi @shundezhang - in the first instance can you please set up the search as per the link above, and evaluate the MyTardis search functionality?

Going back to the original essential requirements that we used to evaluate different systems, can you please consider and advise to what extent MyTardis supports the following?
* Users can search by keyword terms across all metadata fields (i.e. basic search)
* Users can search within select fields corresponding to elements within metadata records. (E.g. for metadata element ‘geloc_country’ there will be a corresponding search field, indexing the values for this element contained within metadata records in the database.)
* Users can search by one or multiple search terms in one query, and can combine multiple search terms using Boolean operators  (i.e. advanced search)
* Users can use facets to refine an initial search or browse query.
* Users can search within files associated to datasets

You may be interested to look at this live example: https://store.synchrotron.org.au/experiment/search/

Can you also advise on what MyTardis is searching on...? I.e. experiment AND dataset metadata? Are results always displayed at Experiment level? If a matching term is found only at the dataset, how is this (if it is at all) highlighted in the results?

@franksc is updating the requirements and will need to know your findings so we can identify gaps. Can you give us an update on things some time next week (week beginning Monday 9 Feb)?

Thank you!

**Comment by shundezhang** [comment]
Hi @millercathy The search function comes from MyTardis can only search on basic experiment data fields, and can't search on dataset fields.  You can try it on http://vm-views-20150202-155450.modc08.ersa.edu.au/experiment/search/

Extra things may need further development.
In MyTardis document solr 1.4 is used but it is very old. We may need to check with the developer to see if a newer version can be used.

**Comment by millercathy** [comment]
Thanks @shundezhang 
So there is clearly a swag of further development required. I think we create these as separate issues. @franksc and I will do this.
@franksc I am going to assign this issue over to you. If there is further investigation required then it can stay open (e..g on solr version) and then close when you're ready.
Cheers.

**Comment by shundezhang** [comment]
After some more research I found that the solr search function in mytardis is currently broken. Solr is used to offer free-text search, like what google has, a single text box and let users enter anything. It comes from Mecat project (https://groups.google.com/forum/#!searchin/tardis-devel/solr/tardis-devel/9Sw9OaELjT8/Hao_cA7TnhgJ). My feeling is mytardis developers don't want to include that in the core and want to have it separated so it can be easily turned on/off. I guess people mostly use database SQL-like search rather than free-text search (by looking at the search page on synchrotron website).

**Comment by millercathy** [comment]
Hi @shundezhang and @craigpatten 
I am about to document search tasks separately... they are all dependent on deciding what search backend we need to meet these requirements.
Can you look at the requirements which will be linked to this issue and use this task to track advice/decision on the search backend - and implementation of that backend. Thanks.

**Comment by shundezhang** [comment]
Code from original mytardis branch is not working for search. While working on a fix, I have updated haystack to 2.3.1 and changed the backend to elasticsearch since the previous code is based on an old version of haystack and solr. Elasticsearch and solr are both based on lucene but elasticsearch has a modern front end and we have used it widely in other projects. As of today search on mytardis models is already working and @franksc will provide mockups and documents on acad models. According to tests on existing code, I anticipate the combination of elasticsearch and haystack will accommodate all known requirements. However, it is highly beneficial if we acquire real data from acad and run tests against it.
